pkg/trace/api/symdb.go

```diff
@@ -0,0 +1,91 @@
+
+import (
+	"fmt"
+	stdlog "log"
+	"net/http"
+	"net/http/httputil"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/google/uuid"
+
+	"github.com/DataDog/datadog-agent/pkg/trace/config"
+	"github.com/DataDog/datadog-agent/pkg/trace/log"
+)
+
+const (
+	// debuggerIntakeURLTemplate specifies the template for obtaining the intake URL along with the site.
+	debuggerIntakeURLTemplate = "https://debugger-intake.%s/api/v2/debugger"
+)
+
+// symDBProxyHandler returns an http.Handler proxying requests to the logs intake. If the logs intake url cannot be
+// parsed, the returned handler will always return http.StatusInternalServerError with a clarifying message.
+func (r *HTTPReceiver) symDBProxyHandler() http.Handler {
+	tags := fmt.Sprintf("host:%s,default_env:%s,agent_version:%s", r.conf.Hostname, r.conf.DefaultEnv, r.conf.AgentVersion)
+	if orch := r.conf.FargateOrchestrator; orch != config.OrchestratorUnknown {
+		tags = tags + ",orchestrator:fargate_" + strings.ToLower(string(orch))
+	}
+	intake := fmt.Sprintf(debuggerIntakeURLTemplate, r.conf.Site)
+	if v := r.conf.SymDBProxy.DDURL; v != "" {
+		intake = v
+	} else if site := r.conf.Site; site != "" {
+		intake = fmt.Sprintf(debuggerIntakeURLTemplate, site)
+	}
+	target, err := url.Parse(intake)
+	if err != nil {
+		log.Criticalf("Error parsing symbol database intake URL %q: %v", intake, err)
+		return symDBErrorHandler(fmt.Errorf("error parsing symbol database intake URL %q: %v", intake, err))
+	}
+	apiKey := r.conf.APIKey()
+	if k := r.conf.SymDBProxy.APIKey; k != "" {
+		apiKey = strings.TrimSpace(k)
+	}
+	transport := newMeasuringForwardingTransport(
+		config.New().NewHTTPTransport(), target, apiKey, r.conf.SymDBProxy.AdditionalEndpoints, "datadog.trace_agent.debugger.", []string{})
+	return newSymDBProxy(r.conf, transport, tags)
+}
+
+// symDBErrorHandler always returns http.StatusInternalServerError with a clarifying message.
+func symDBErrorHandler(err error) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		msg := fmt.Sprintf("SymDB Proxy is OFF: %v", err)
+		http.Error(w, msg, http.StatusInternalServerError)
+	})
+}
+
+// newSymDBProxy returns a new httputil.ReverseProxy proxying and augmenting requests with headers containing the tags.
+func newSymDBProxy(conf *config.AgentConfig, transport http.RoundTripper, tags string) *httputil.ReverseProxy {
+	cidProvider := NewIDProvider(conf.ContainerProcRoot)
+	logger := log.NewThrottled(5, 10*time.Second) // limit to 5 messages every 10 seconds
+	return &httputil.ReverseProxy{
+		Director:  getSymDBDirector(tags, cidProvider, conf.ContainerTags),
+		ErrorLog:  stdlog.New(logger, "symdb.Proxy: ", 0),
+		Transport: transport,
+	}
+}
+
+func getSymDBDirector(tags string, cidProvider IDProvider, containerTags func(string) ([]string, error)) func(*http.Request) {
+	return func(req *http.Request) {
+		req.Header.Set("DD-REQUEST-ID", uuid.New().String())
+		req.Header.Set("DD-EVP-ORIGIN", "agent-symdb")
```

Comment:
What are the requirements for this value, do we need to register it anywhere?

Reply:
It's part of the v2 intake and it's used for observability / debugging purposes. There is no need to register it.